### PR TITLE
Deck: pr history supports `revision` field from pod-utils

### PR DESCRIPTION
**Pull request by ibzib**
We will also probably have to add `revision` to the Spyglass metadata viewer.

/assign @krzyzacy

### CODE REVIEW

File: prow/cmd/deck/job_history.go [hunk 1]
```diff
@@ -297,8 +297,9 @@ func getBuildData(bucket storageBucket, dir string) (buildData, error) {
 		return b, fmt.Errorf("failed to read started.json: %v", err)
 	}
 	b.Started = time.Unix(started.Timestamp, 0)
-	commitHash, err := getPullCommitHash(started.Pull)
-	if err == nil {
+	if started.Revision != "" {
```
Comment (krzyzacy) [inline prow/cmd/deck/job_history.go]: Maybe worth check both started and finished? We actually write revision in finished.json - (https://github.com/kubernetes/test-infra/blob/master/prow/sidecar/run.go#L189-L193) but technically you can also write it in started.

**Comment by stevekuznetsov** [comment]
/hold

Can we please, please, please:

1. Write a design document for where we _WANT_ data to live and what form it should take
2. Think clearly and critically about how to do this in a simple and straightforward way

A field like "revision" is much less useful than the entire refs struct from the ProwJob. We should use ProwJobSpec wherever possible.

Having to look in two files optionally is awful. Let's not organically re-build Gubernator. That's not why we decided to retire it.

**Comment by BenTheElder** [comment]
Actually a major reason is being able to ship all of Prow the same way, on
k8s. The revision field is useful and this is a small step. Please do not
block incremental improvements.

**Comment by stevekuznetsov** [comment]
Where's the larger DD and vision? I have strong concerns about this ... the current `started.json` and `finished.json` uses are very crusty and impossible to untangle. This is a small step in a larger walk in the wrong direction.

**Comment by BenTheElder** [comment]
There's no DD for +8/-7 ... I don't have the original spyglass doc handy but we've both read it.

> This is a small step in a larger walk in the wrong direction.

This is such an incredibly small step that if that happens to be true we can simply step backwards...

File: prow/cmd/deck/job_history.go (hunk 1 above)
Comment (ibzib) [reply prow/cmd/deck/job_history.go]: done

**Comment by stevekuznetsov** [comment]
The fact that it's small has no bearing on my comments. This is net-new and uses `pod-utils` and `deck`. If we want `deck` and `spyglass` to display information about the refs that were tested, why not upload the entire `ProwJobSpec` or at the very least the `Refs` from the `pod-utils` into a _known_ and stable location, then intelligently use the data in those structs? We have the power to do this right, in this PR. Why do we need the current impl?

**Comment by BenTheElder** [comment]
This PR merely displays data we're already uploading, so people can view that data today.
It is small, pragmatic, and useful, and even has tests.

If you'd like to come back later and write something more sophisticated, that sounds great. This PR will not prevent that.
/hold cancel

**Comment by krzyzacy** [comment]
@stevekuznetsov I would have a place to hold a simple SHA from the main ref, for downstream to consume, rather than we store a list of Ref objects, and then let downstream to figure out what to display. Things like gubernator won't need to know how a Ref object should look like.

also, I mean, https://github.com/kubernetes/test-infra/pull/10059#discussion_r236739249 :-) Or you want to dump the downstream spec for now? :-)

**Comment by stevekuznetsov** [comment]
We're not talking about Gubernator. We're talking about Prow. All of these components are internal to Prow. Let's make the useful and beneficial choice here, not the simple one that copies cruft from Gubernator. Need I link to the (three year old) issue where I was first trying to figure out WTF the format of these files was, and the fact that nobody was ever able to give me a clear answer?

**Comment by krzyzacy** [comment]
@stevekuznetsov the problem I see is we have two definition of started/finished struct within prow, we should merge these and document what does each field mean.

We probably also need to keep backwards compatibility to bootstrap (the no-clear-answer version) but we can retire that soon.

**Comment by ibzib** [comment]
> we have two definition of started/finished struct within prow, we should merge these and document what does each field mean

100% agree. There's even another definition in testgrid 😨😨😨 
https://github.com/ibzib/test-infra/blob/5b57a7ea791e09620380eae53cd51870390ecf2f/testgrid/cmd/updater/main.go#L117-L142

**Pull request merged**